### PR TITLE
refactor!: use new icons approach

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
Closes #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/components/icon/examples/3/index.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <tui-icon
-    background="tuiIconInfoLarge"
-    icon="tuiIconInfoLarge"
+    background="tuiIconInfo
```
Comment (waterplea) [inline projects/demo/src/modules/components/icon/examples/3/index.html:2]: Something's wrong here

File: projects/demo/src/modules/components/icons/icons.tokens.ts [hunk 2]
```diff
@@ -2,9 +2,9 @@ import {inject} from '@angular/core';
 import type {TuiStringHandler} from '@taiga-ui/cdk';
 import {tuiCreateTokenFromFactory} from '@taiga-ui/cdk';
 import {TUI_SVG_OPTIONS} from '@taiga-ui/core';
-import * as allIcons from '@taiga-ui/icons';
+import {TUI_ALL_ICONS} from '@taiga-ui/icons';
```
Comment (waterplea) [inline projects/demo/src/modules/components/icons/icons.tokens.ts:5]: Should we still compile a typescript dictionary of all the icons? Doesn't seem like it makes sense now that you cannot pass source code to the icon component. Preloading icons in bundle would work differently now.

File: projects/icons/scripts/prepare-lucide-icons.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        const content = fs.readFileSync(path.join(src, filename), 'utf-8');
+
+        fs.writeFileSync(
+            path.join(dest, processName(filename)),
+            content.replace('stroke-width="2"', 'stroke-width="3"'),
+        );
+    });
+})();
+
+function processName(name: string, postfix = ''): string {
```
Comment (waterplea) [inline projects/icons/scripts/prepare-lucide-icons.ts:18]: Just a thought for future: I think we should minimize icon processing, probably in the next PR — we can keep the names as they are. I also wonder if it's possible to created filled version of the icons from lucide, would setting stroke to transparent and fill to current color do the trick?

File: projects/demo/src/modules/components/icons/icons.tokens.ts (hunk 2 above)
Comment (waterplea) [reply projects/demo/src/modules/components/icons/icons.tokens.ts:5]: I dropped it in my PR, I don't think if we maintain name equivalence we need a separate page to list all the 1500 icons.

Comment (waterplea) [reply projects/demo/src/modules/components/icons/icons.tokens.ts:5]: That's why I also want to drop file/name processing and just write it like `iconLeft="@tui.align-horizontal-distribute-start"` so you can just copy icon name from Lucide docs.

File: projects/icons/scripts/prepare-lucide-icons.ts (hunk 3 above)
Comment (vladimirpotekhin) [reply projects/icons/scripts/prepare-lucide-icons.ts:21]: > I also wonder if it's possible to created filled version of the icons from lucide, would setting stroke to transparent and fill to current color do the trick?

Seems like it works (example with apple icon). But why do we need to create filled versions? This is a little contrary to minimizing preprocessing

Снимок экрана 2024-06-14 в 10 27 23
Снимок экрана 2024-06-14 в 10 27 48

**Review by waterplea — APPROVED** [review]

File: projects/icons/scripts/prepare-lucide-icons.ts (hunk 3 above)
Comment (vladimirpotekhin) [reply projects/icons/scripts/prepare-lucide-icons.ts:21]: I will add filled icons in next PR

**Pull request merged**